index.js

```diff
@@ -149,42 +201,62 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+// push a number to line array 
+// return message 
+let counting = 0
+
+function takeATicketNumber (line) {
+  counting ++;
+  line.push(counting);
+  return `Welcome. You are ticket number ${counting}`
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
-// console.log("%c----------", "color: red") 
+console.log(nowServing(line))
+// => `Currently serving 2.`
+console.log("%c----------", "color: red") 
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
 
 // ***** Scope & Closures - Question 2 *****
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+function ticketNumberGeneratorFunc() {
+  let count = 0;
+  function takeATicketNumber (line) {
+    count ++;
+    line.push(count);
+    return `Welcome. You are ticket number ${counting}`
+  }
+  return takeATicketNumber;
+}
```

Comment:
Great job getting this closure working! It's a tricky thing to understand, but once you do, you can do all kinds of neat things in JS.